pkgs/servers/home-assistant/custom-components/README.md

```diff
@@ -0,0 +1,37 @@
+# Packaging guidelines
+
+## buildCustomComponent
+
+Custom components should be packaged using the `buildCustomComponent`
+function, that is provided by `home-assistant`. It builds upon
+`buildPythonPackage` but uses a custom install and check phase.
+
+Python runtime dependencies that the component requires go into
+`propagatedBuildInputs`.
```

Comment:
does that mean my package should have `python3` as an input/parameter to get `python3.pkgs.mydependency`, or is there a way to get 'the python used by homeassistant'?

Reply:
You can consume them in the argument list.

Reply:
I'm not sure I understand what you mean - could you give an example?

Reply:
```nix
{ lib
, home-assistant
, fetchFromGitHub
# your python dependencies go here
, <here>
}:

Reply:
hmm, I did try that, but got e.g. `Function called without required argument "aio-geojson-client"`.

I actually want to import libgpiod, but that might be a bit of a special case: there is both a C library with that name in `all-packages.nix` and its python bindings in `python-packages.nix`.

Reply:
Tested like this the result now propagates aio-geojson-client.

```patch
diff --git a/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix b/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix
index 94194b5ec8a4..0c2d223826ac 100644
--- a/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix
+++ b/pkgs/servers/home-assistant/custom-components/prometheus-sensor/default.nix
@@ -1,6 +1,7 @@
 { lib
 , fetchFromGitHub
 , home-assistant
+, aio-geojson-client
 }:
 
 home-assistant.buildCustomComponent rec {
@@ -16,6 +17,10 @@ home-assistant.buildCustomComponent rec {
 
   dontBuild = true;
 
+  propagatedBuildInputs = [
+    aio-geojson-client
+  ];
+
   meta = with lib; {
     changelog = "https://github.com/mweinelt/ha-prometheus-sensor/blob/${version}/CHANGELOG.md";
     description = "Import prometheus query results into Home Assistant";
```

```
$ cat result/nix-support/propagated-build-inputs
/nix/store/z5781bknkpj7j1rmyh29c0yzqqp9cjbx-python3.11-aio-geojson-client-0.18 /nix/store/ffll6glz3gwx342z0ch8wx30p5cnqz1z-python3-3.11.5 %      
```

Reply:
ah right, because you're using `python.pkgs.callPackage` for the in-tree custom components (https://github.com/NixOS/nixpkgs/pull/160346/files#diff-843aa716f56e9c6c540c99c5461a90a32efd9422be9a251dcf67244a4b8d6421R570).

Looks like for an out-of-tree custom component I can take `home-assistant.python` to get the 'right' python :+1: